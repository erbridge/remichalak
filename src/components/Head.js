import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { Helmet, HelmetProvider } from 'react-helmet-async';
import { Route } from 'react-router-dom';
import routePropType from '../prop-types/routePropType';

class Head extends Component {
  static propTypes = {
    routes: PropTypes.arrayOf(routePropType).isRequired,
  };

  render() {
    const { routes } = this.props;

    return (
      <HelmetProvider>
        {routes.map((route, index) => (
          <Route
            key={route.name || index}
            path={route.path}
            exact={route.exact}
          >
            <Helmet>
              <title>
                {route.title ? `${route.title} | ` : ''}R. E. Michalak
              </title>
            </Helmet>
          </Route>
        ))}
      </HelmetProvider>
    );
  }
}

export default Head;
