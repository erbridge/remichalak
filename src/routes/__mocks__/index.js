import React from 'react';

export const topRoutes = [
  {
    name: 'test-internal',
    link: 'Test Internal',
    path: '/test',
    exact: true,
    title: 'Test Internal Title',
    content: [() => <div>1</div>, [() => <div>2-1</div>, () => <div>2-2</div>]],
  },
  {
    name: 'test-external',
    link: 'Test External',
    path: 'http://example.com',
    external: true,
  },
];

export const mainRoutes = [];

export default [...topRoutes, ...mainRoutes];
