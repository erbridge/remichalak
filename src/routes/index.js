import about_1 from '../pages/about/1.mdx';
import about_2_1 from '../pages/about/2-1.mdx';
import about_2_2 from '../pages/about/2-2.mdx';
import contact from '../pages/contact.mdx';

export const topRoutes = [
  {
    name: 'about',
    link: 'About',
    path: '/',
    exact: true,
    title: about_1.title,
    content: [about_1, [about_2_1, about_2_2]],
  },
  {
    name: 'shop',
    link: 'Shop',
    path: 'https://society6.com/remichalak',
    external: true,
  },
  {
    name: 'contact',
    link: 'Contact',
    path: '/contact',
    title: contact.title,
    content: contact,
  },
];

export const mainRoutes = [];

export default [...topRoutes, ...mainRoutes];
