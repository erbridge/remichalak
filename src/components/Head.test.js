import { shallow } from 'enzyme';
import React from 'react';
import routes from '../routes';
import Head from './Head';

it('renders correctly', () => {
  expect(shallow(<Head routes={routes} />)).toMatchSnapshot();
});
