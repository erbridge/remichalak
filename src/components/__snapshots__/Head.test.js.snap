// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<HelmetProvider
  context={Object {}}
>
  <Route
    exact={true}
    key="about"
    path="/"
  >
    <Helmet
      defer={true}
      encodeSpecialCharacters={true}
    >
      <title>
        R. E. Michalak
      </title>
    </Helmet>
  </Route>
  <Route
    key="shop"
    path="https://society6.com/remichalak"
  >
    <Helmet
      defer={true}
      encodeSpecialCharacters={true}
    >
      <title>
        R. E. Michalak
      </title>
    </Helmet>
  </Route>
  <Route
    key="contact"
    path="/contact"
  >
    <Helmet
      defer={true}
      encodeSpecialCharacters={true}
    >
      <title>
        R. E. Michalak
      </title>
    </Helmet>
  </Route>
</HelmetProvider>
`;
